beacon-chain/blockchain/service.go

```diff
@@ -16,26 +16,30 @@ var log = logrus.WithField("prefix", "blockchain")
 // ChainService represents a service that handles the internal
 // logic of managing the full PoS beacon chain.
 type ChainService struct {
-	ctx                              context.Context
-	cancel                           context.CancelFunc
-	beaconDB                         *database.DB
-	chain                            *BeaconChain
-	web3Service                      *powchain.Web3Service
-	latestBeaconBlock                chan *types.Block
-	processedBlockHashes             [][32]byte
-	processedActiveStateHashes       [][32]byte
-	processedCrystallizedStateHashes [][32]byte
+	ctx                                    context.Context
+	cancel                                 context.CancelFunc
+	beaconDB                               *database.DB
+	chain                                  *BeaconChain
+	web3Service                            *powchain.Web3Service
+	canonicalBlockAnnouncement             chan *types.Block
+	canonicalCrystallizedStateAnnouncement chan *types.CrystallizedState
+	latestBeaconBlock                      chan *types.Block
+	processedBlockHashes                   [][32]byte
```

Comment:
What are these used for? Are we keeping a copy of all of the blocks we've seen in memory? 

Some comments on these fields would be helpful for reader understanding.